INBOX-344: force HTTPS when following Link headers

rupa opened the pull request:
When the Client endpoint is set to an HTTPS URL, rewrite any HTTP Link
URL to HTTPS.

Commit:
INBOX-344: force HTTPS when following Link headers

When the Client endpoint is set to an HTTPS URL, rewrite any HTTP Link
URL to HTTPS.

Commit:
INBOX-344: shim in an explicit pagination test with HTTP endpoint

mburtless commented on rest/headers.go:
```diff
@@ -38,7 +38,7 @@ func (l Links) Next() string {
 // ParseLink parses a Link header value into a Links, mainly cribbed from:
 // https://github.com/peterhellberg/link/blob/master/link.go
-func ParseLink(s string) Links {
+func ParseLink(s string, forceHTTPS bool) Links {
```
Could use an updated description that describes the effect of `forceHTTPS`

mburtless reviewed (approved):
LGTM. I think we should document the new param in `ParseLink` though, as that's exported.

Commit:
INBOX-344: document the new param to ParseLinks

Proposed commit message for squash/merge of this PR:
====================================================

INBOX-344: force HTTPS when following Link headers

When the Client endpoint is set to an HTTPS URL, rewrite any HTTP
Link URLs to HTTPS.

Note a small but potentially breaking change:

This commit changes the arguments to `ParseLinks`. This is a
very recently added function that should not need to be explicity
called in normal SDK usage, but _is_ exported for convenience.
